src/libs/actions/App.ts

```diff
@@ -596,7 +588,7 @@ function setPreservedUserSession(session: OnyxTypes.Session) {
     Onyx.set(ONYXKEYS.PRESERVED_USER_SESSION, session);
 }
 
-function clearOnyxAndResetApp(shouldNavigateToHomepage?: boolean) {
+function clearOnyxAndResetApp(preservedUserSession: OnyxEntry<OnyxTypes.Session> | undefined, shouldNavigateToHomepage?: boolean) {
     // The value of isUsingImportedState will be lost once Onyx is cleared, so we need to store it
```

Comment:
I'm actually rethinking this approach. Considering that the view won't really change the `preservedUserSession`, why are we deciding to send it from the view to this method instead of just changing the dependency in this file?